src/lzmaio.hpp

```diff
@@ -0,0 +1,127 @@
+#ifndef LZMAIO_HPP
+#define LZMAIO_HPP
+
+#include <stdlib.h>
+#include <stdio.h>
+#include <string.h>
+#include <lzma.h>
+
+typedef struct {
+	static const int BUFFER_SIZE = 4096;
+
+	lzma_stream decode;
+	lzma_stream encode;
+	FILE* fp;
+
+	size_t in_size;
+	uint8_t in[BUFFER_SIZE];
+
+	size_t out_size;
+	uint8_t out[BUFFER_SIZE];
+} LZMAFILE;
+
+LZMAFILE* lzmaopen(const char* filepath, const char* mode) {
+	LZMAFILE* file = (LZMAFILE*)malloc(sizeof(LZMAFILE));
```

Comment:
be careful not to cause a memory leak, the world of memory allocation is dangerous